src/service.js

```diff
@@ -217,6 +219,43 @@ export function fromClass(win, id, constructor) {
   return getServiceInternal(win, win, id, undefined, constructor);
 }
 
+/**
+ * Registers a service given a class to be used as implementation.
+ * @param {!Window} win
+ * @param {string} id of the service.
+ * @param {function(new:Object, !Window)} constructor
+ * @param {boolean=} opt_instantiate Whether to immediately create the service
+ */
+export function registerServiceConstructor(win,
+                                           id,
+                                           constructor,
+                                           opt_instantiate) {
+  win = getTopWindow(win);
+  registerServiceInternal(win, win, id, constructor);
+  if (opt_instantiate) {
+    getServiceInternal(win, win, id);
+  }
+}
+
+/**
+ * Returns a service and registers it given a class to be used as
+ * implementation.
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
+ * @param {string} id of the service.
+ * @param {function(new:Object, !./service/ampdoc-impl.AmpDoc)} constructor
+ * @param {boolean=} opt_instantiate Whether to immediately create the service
+ */
+export function registerServiceForDoc(nodeOrDoc,
```

Comment:
`registerServiceBuilderForDoc` to be consistent with whatever you rename the method above to.